### PR TITLE
[Sprint: 48] XD-3003 Adding support for Sqoop's metastore and job command

### PR DESCRIPTION
**Pull request by trisberg**
- job --exec requires the use of --password-file or the job will prompt for a password and execution will hang

### CODE REVIEW

**Comment by trisberg** [comment]
The metastore must run in PostgreSQL and I recommend creating the tables manually:

```
CREATE TABLE
    SQOOP_ROOT
    (
        version INTEGER,
        propname CHARACTER VARYING(128) NOT NULL,
        propval CHARACTER VARYING(256),
        UNIQUE (version, propname)
    );
CREATE TABLE
    SQOOP_SESSIONS
    (
        job_name CHARACTER VARYING(64) NOT NULL,
        propname CHARACTER VARYING(128) NOT NULL,
        propval CHARACTER VARYING(1024),
        propclass CHARACTER VARYING(32) NOT NULL,
        UNIQUE (job_name, propname, propclass)
    );
INSERT INTO sqoop_root (version, propname, propval) VALUES (null, 'sqoop.hsqldb.job.storage.version', '0');
INSERT INTO sqoop_root (version, propname, propval) VALUES (0, 'sqoop.hsqldb.job.info.table', 'SQOOP_SESSIONS');
```

File: src/docs/asciidoc/Batch-Jobs.asciidoc [hunk 1]
```diff
@@ -725,6 +726,66 @@ job create sqoopComplexArgs1 --definition "sqoop --command=import --args='--tabl
...
+
+----
+xd>job create job1create --definition "sqoop --command=job --args='--create job1 -- import --table PETS --incremental append --check-column ID --last-value 0 --connect jdbc:hsqldb:hsql://localhost:9001/test --username sa --password-file /xd/hsql.password --target-dir /xd/job1 --num-mappers 1'" --deploy
+xd>job launch job1create
+----
+
+Here is an example of executing the predefined Sqoop job using Spring XD's `sqoop` job:
+
+----
+xd>job create job1exec --definition "sqoop --command=job --args='--exec job1" --deploy
```
Comment (jvalkeal) [inline src/docs/asciidoc/Batch-Jobs.asciidoc]: Missing terminating ' character. I'll add that during a merge.

File: src/docs/asciidoc/Batch-Jobs.asciidoc [hunk 2]
```diff
@@ -725,6 +726,66 @@ job create sqoopComplexArgs1 --definition "sqoop --command=import --args='--tabl
...
+- `sqoop.metastore.client.autoconnect.username`
+- `sqoop.metastore.client.autoconnect.password`
+
+NOTE: In addition to the above configurations you need to use a `--password-file` option when creating the Sqoop job definitions. If you don't then Sqoop will prompt for a password
+as Spring XD runs the job. This will cause the job to hang.
+
+Here is an example of defining a Sqoop job using Spring XD's `sqoop` job:
+
+----
+xd>job create job1create --definition "sqoop --command=job --args='--create job1 -- import --table PETS --incremental append --check-column ID --last-value 0 --connect jdbc:hsqldb:hsql://localhost:9001/test --username sa --password-file /xd/hsql.password --target-dir /xd/job1 --num-mappers 1'" --deploy
```
Comment (jvalkeal) [inline src/docs/asciidoc/Batch-Jobs.asciidoc]: I used gpdb as db. Below is a command what sqoop would run when job is later executed. It fails if table is in caps and later we get same trouble with column name.

```
create table PETS (id integer, name text) distributed by (id);

gpadmin=# SELECT t.* FROM "PETS" AS t LIMIT 1;
ERROR:  relation "PETS" does not exist
LINE 1: SELECT t.* FROM "PETS" AS t LIMIT 1;
                        ^
gpadmin=# SELECT t.* FROM "pets" AS t LIMIT 1;
 id | name 
----+------
  4 | pet4

gpadmin=# select count(*) from pets;
 count 
-------
     5
(1 row)

gpadmin=# select count(*) from PETS;
 count 
-------
     5
(1 row)

gpadmin=# select count(*) from "PETS";
ERROR:  relation "PETS" does not exist
LINE 1: select count(*) from "PETS";
                             ^
gpadmin=# select count(*) from "pets";
 count 
-------
     5
(1 row)
```
Anyway just adding note here.

**Comment by jvalkeal** [comment]
Other than that, looks good, merging.

**Comment by jvalkeal** [comment]
Merged per https://github.com/spring-projects/spring-xd/commit/e6f825801ed69aa6ee5baec656a883e07d714e71, closing.

**Pull request closed without merging**